calcom/cal.com
# Fix/unpaid/unconfirmed/#2170

## What does this PR do?



Fixes #2170

- When you disconnect the stripe credential and you book paid event type, it shows you "ERROR [api] book:user Creating payment failed" but it creates the booking. So to fix it, I added in /apps/web/pages/api/book/events, a filter that validates if there is any stripe_payment credential for this user, if the stripe_payment credential exists then it creates the booking, and it just saved the booking in Calendar if it is paid and confirmed. 

<img width="316" alt="image" src="https://user-images.githubusercontent.com/19562383/161136214-caecf4b4-bad8-4f57-8b13-5fd465e5d19d.png">

And shows the same message but it does not create the booking
<img width="260" alt="image" src="https://user-images.githubusercontent.com/19562383/161140048-c299fea5-f5fc-4ea1-8c23-5aaefb28020f.png">


- For bookings that do not have payment, when it creates the booking, I assigned paid value to true, and I added in  /apps/web/pages/api/book/events to take the calendarBusyTimes a filter that only bookings that are confirmed and have paid are true, they will take the availability. So if you book a paid event at 9:00 am (and stripe credential exists) it creates the booking and you can see it on/Booking page, and then another person books the same paid event at the same time, he can book it because the first one is not confirmed, even it is paid. 

<img width="605" alt="image" src="https://user-images.githubusercontent.com/19562383/161135703-80adb771-1d8a-4df3-a836-1862feeb26dd.png">

- When you have bookings from a paid event and these are not confirmed or paid, and you disconnect the stripe credential, these bookings will be rejected.

## Type of change




## How should this be tested?



Connect a Stripe account
Create a priced event type
Remove Stripe account 
Try to book paid event type
Go to /bookings and you going to see the booking was not created.
